File: frontend/src/shared/types/tasks.ts
export type LearningTask = {
  task_id: string;
  student_id: string;
  title: string;
  source: string;
  status: "todo" | "doing" | "done" | string;
  priority: "high" | "medium" | "low" | string;
  due_date: string;
  evidence_required: string;
  progress: number;
};

export type TaskListResponse = {
  student_id: string;
  total: number;
  completed: number;
  tasks: LearningTask[];
};

export type ReviewResponse = {
  review_id: string;
  student_id: string;
  period: string;
  summary: string;
  completed_count: number;
  risk: string;
  next_tasks: LearningTask[];
  ai_generated: boolean;
};

export type AgentTaskCreateRequest = {
  task_type: string;
  owner_id: string;
  input?: Record<string, unknown>;
};

export type AgentTaskStatus = {
  task_id: string;
  task_type: string;
  status: string;
  owner_id?: string | null;
  input: Record<string, unknown>;
  state: Record<string, unknown>;
  result_ref?: string | null;
  error_message?: string | null;
  created_at: string;
  updated_at: string;
};

export type AgentTaskActionResponse = {
  task: AgentTaskStatus;
  message: string;
};

File: backend/app/schemas/tasks.py
from __future__ import annotations

from pydantic import BaseModel, Field


class LearningTask(BaseModel):
    task_id: str
    student_id: str
    title: str
    source: str
    status: str
    priority: str
    due_date: str
    evidence_required: str
    progress: int


class TaskListResponse(BaseModel):
    student_id: str
    total: int
    completed: int
    tasks: list[LearningTask]


class SaveTaskRequest(BaseModel):
    student_id: str = "student_001"
    title: str
    source: str = "manual"
    priority: str = "medium"
    due_date: str = "2026-07-12"
    evidence_required: str = "提交学习记录或项目产物"


class ReviewRequest(BaseModel):
    student_id: str = "student_001"
    period: str = "本周"
    completed_task_ids: list[str] = Field(default_factory=list)
    notes: str | None = None


class ReviewResponse(BaseModel):
    review_id: str
    student_id: str
    period: str
    summary: str
    completed_count: int
    risk: str
    next_tasks: list[LearningTask]
    ai_generated: bool = True


class AgentTaskCreateRequest(BaseModel):
    task_type: str = "assignment_analysis"
    owner_id: str = "student_001"
    input: dict = Field(default_factory=dict)


class AgentTaskStatus(BaseModel):
    task_id: str
    task_type: str
    status: str
    owner_id: str | None = None
    input: dict = Field(default_factory=dict)
    state: dict = Field(default_factory=dict)
    result_ref: str | None = None
    error_message: str | None = None
    created_at: str
    updated_at: str


class AgentTaskActionResponse(BaseModel):
    task: AgentTaskStatus
    message: str
